dotnet/src/webdriver/Script.cs

```diff
@@ -0,0 +1,99 @@
+// <copyright file="Script.cs" company="WebDriver Committers">
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Threading.Tasks;
+using WebDriverBiDi;
+using WebDriverBiDi.Log;
+using WebDriverBiDi.Session;
+
+namespace OpenQA.Selenium
+{
+    /// <summary>
+    /// Provides a mechanism to access WebDriverBiDi events in script and log domains.
+    /// </summary>
+    [Obsolete("This class is in beta and may change in future releases.")]
+    internal class Script : IScript
+    {
+        private readonly BiDiDriver biDiDriver;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Script"/> class
+        /// </summary>
+        /// <param name="driver">Driver in use</param>
+        [Obsolete("This class is in beta and may change in future releases.")]
+        public Script(WebDriver driver)
+        {
+            this.biDiDriver = driver.BiDiDriver;
+            this.biDiDriver.Log.EntryAdded += OnEntryAdded;
+        }
+
+        /// <summary>
+        /// Add and remove handlers for console messages.
+        /// </summary>
+        [Obsolete("This event is in beta and may change in future releases.")]
```

Comment:
This is something of an abuse of the `Obsolete` attribute. No, the .NET API doesn't have anything like the `@Beta` annotation, and I get what you're going for (an attribute that automatically warns when compiling). But seeing this warning will be surprising to users, because the warning code will be the one that means "You're using a method that may be removed."

Reply:
Oh yeah, I'm abusing it. The question is whether we have to commit to this API before merging/releasing. We probably are ok to commit to it, so I can remove it.